includes/class-wcdn-writepanel.php

```diff
@@ -161,60 +161,75 @@ public function my_bulk_action_handler( $redirect_to, $doaction, $post_ids ) {
 			}
 
 			// security check.
-			if ( isset( $_GET['page'] ) && 'wc-orders' === $_GET['page'] ) {
-				check_admin_referer( 'bulk-orders' );
-			} else {
-				check_admin_referer( 'bulk-posts' );
-			}
-
-			// get referrer.
-			if ( ! wp_get_referer() ) {
-				return $redirect_to;
-			}
+			check_admin_referer( 'bulk-orders' );
 
-			// filter the referer args.
-			$referer_args = array();
-			parse_str( wp_parse_url( wp_get_referer(), PHP_URL_QUERY ), $referer_args );
+			// Get the total number of post IDs.
+			$total     = count( $post_ids );
+			$print_url = htmlspecialchars_decode( wcdn_get_print_link( $post_ids, $template_type ) );
 
-			// set the basic args for the sendback.
-			$args = array();
-			if ( isset( $referer_args['post_status'] ) ) {
-				$args = wp_parse_args( array( 'post_status' => $referer_args['post_status'] ), $args );
-			}
-			if ( isset( $referer_args['paged'] ) ) {
-				$args = wp_parse_args( array( 'paged' => $referer_args['paged'] ), $args );
-			}
-			if ( isset( $referer_args['orderby'] ) ) {
-				$args = wp_parse_args( array( 'orderby' => $referer_args['orderby'] ), $args );
-			}
-			if ( isset( $referer_args['order'] ) ) {
-				$args = wp_parse_args( array( 'orderby' => $referer_args['order'] ), $args );
-			}
-
-			if ( isset( $referer_args['post_type'] ) ) {
-				$args = wp_parse_args( array( 'post_type' => $referer_args['post_type'] ), $args );
-			}
+			// WooCommerce orders page URL.
+			$orders_page_url = admin_url( 'admin.php?page=wc-orders' );
```

Comment:
@gaminikhil We should consider non HPOS orders page also here. You can add a condition for that.